### ISSUE TITLE
Cache by URL - Grav CMS Dynamic PHP

### ISSUE THREAD

**Issue by STnetwork** [GoogleChrome/workbox#830]
Welcome! Please use this template for reporting bugs or requesting features. For questions about using Workbox, the best place to ask is Stack Overflow, tagged with `[workbox]`: https://stackoverflow.com/questions/ask?tags=workbox

**Library Affected**:
*workbox-sw, workbox-build, etc.*

**Browser & Platform**:
*E.g. Google Chrome v51.0.1 for Android, or "all browsers".*

**Issue or Feature Request Description**:
*Your request here.*

*When reporting bugs, please include relevant JavaScript Console logs and links to public URLs at which the issue could be reproduced.*

Hi,

My website is based on Grav CMS, it's an application PHP, so it's completely dynamic I don't have any static file instead of the style, so I would like to cache for offline my website by the URL is it possible ?

Thanks

**Comment by gauntface** [comment]
@STnetwork yes it's possible, your server will need to create some kind of hash for each url you cache and use that to precache.

So on your server you'd need to create a javascript array like:

```javascript
const precacheList = [
  {url: '/' revision: '1234567'},
  {url: '/styles/index.css', revision: '1234567'},
  {url: '/scripts/index.js', revision: '1234567'}.
];
```

Then you can use that in your SW:

```javascript
importScripts('<Path to workbox-sw>');

<Inject precache list here>

const workbox = new WorkboxSW();
workbox.precache(precacheList);
```

**Comment by STnetwork** [comment]
I am going to try it ! thanks

**Comment by STnetwork** [comment]
Hey @gauntface 

I did it, but I can only access offline to the js and css files, but not the root site and the rest site too.

```
importScripts('workbox-sw.prod.v2.0.3.js');

/**
 * DO NOT EDIT THE FILE MANIFEST ENTRY
 *
 * The method precache() does the following:
 * 1. Cache URLs in the manifest to a local cache.
 * 2. When a network request is made for any of these URLs the response
 *    will ALWAYS comes from the cache, NEVER the network.
 * 3. When the service worker changes ONLY assets with a revision change are
 *    updated, old cache entries are left as is.
 *
 * By changing the file manifest manually, your users may end up not receiving
 * new versions of files because the revision hasn't changed.
 *
 * Please use workbox-build or some other tool / approach to generate the file
 * manifest which accounts for changes to local files and update the revision
 * accordingly.
 */
const precacheList = [
  {url: '/', revision: '1234567'},
  {url: '/user/themes/haywire/dist/css/app.css', revision: '1234567'},
  {url: '/user/themes/haywire/dist/js/app.js', revision: '1234567'}
];

const workboxSW = new self.WorkboxSW();
workboxSW.precache(precacheList);
```

**Comment by gauntface** [comment]
Your service worker should be at the root of your site (i.e. https://stnetwork.fr/sw.js)

**Comment by STnetwork** [comment]
@gauntface 
Yes ! I put it on the root of my site, so now the root of my site is cached offline ! but only the root, so I guess I have to do something else in my `sw.js` for cache the whole site offline ?

**Comment by gauntface** [comment]
@STnetwork you'll need to write the code to find the files you want to cache and add them to the list.

**Comment by STnetwork** [comment]
Ok it's good for me, last question how I can generate this sw conf with `workbox-webpack-plugin` ?

**Comment by gauntface** [comment]
Best bet would be to look here: https://workboxjs.org/get-started/webpack.html

If you get stuck with that, probably best to raise an issue on stackoverflow.

**Comment by diazwatson** [comment]
Hi @STnetwork did you manage to get workbox to work with Grav?
